doc/auto_techsupport_and_coredump_mgmt.md

```diff
@@ -0,0 +1,398 @@
+The naming format and compression is governed by the script `/usr/local/bin/coredump-compress`.
+
+Where `<comm>` value in the command name associated with a process. comm value of a running process can be read from `/proc/[pid]/comm` file
+
+## 4. Schema Additions
+
+### Config DB
+
+#### AUTO_TECHSUPPORT|GLOBAL
+```
+key = "AUTO_TECHSUPPORT|global"
+state = enabled|disabled;                   # Enable this to make the Techsupport Invocation event driven based on core-dump generation
+rate_limit_interval = 300;                  # Minimum Time in seconds, between two successive techsupport invocations.
+                                              Manual Invocations will be considered as well in the cooloff calculation
+max_techsupport_size = 10;                  # A perentage value should be specified. 
+                                              This signifies maximum Size to which /var/dump/ directory can be grown until. 
+                                              The actual value in bytes is calculate based on the available space in the filesystem hosting /var/dump
+                                              When the limit is crossed, the older techsupport dumps are incrementally deleted                         
+max_core_size = 5;                          # A perentage value should be specified. 
+                                              This signifies maximum Size to which /var/core directory can be grown until. 
+                                              The actual value in bytes is calculate based on the available space in the filesystem hosting /var/core
+                                              When the limit is crossed, the older core files are incrementally deleted
+since = "2 days ago";                       # This limits the auto-invoked techsupport to only collect the logs & core-dumps generated since the time provided.
+                                              Any valid date string of the formats specified here (https://www.gnu.org/software/coreutils/manual/html_node/Date-input-formats.html) 
+                                              can be used. If this value is not explicitly configured or a non-valid string is provided, a default value of "2 days ago" is used.      
+```
+
+#### AUTO_TECHSUPPORT_FEATURE|<feature_name>
+```
+state =  enabled|disabled;              # Enable auto techsupport invocation on the critical processes running inside this feature
+rate_limit_interval = 600;                # Rate limit interval for the corresponding feature. Configure 0 to explicitly disable
+```
+
+                           
+#### YANG Models
+
+```
+module sonic-auto_techsupport {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-auto_techsupport";
+    prefix auto_techsupport;
+
+    description "Event Driven Techsupport & CoreDump Mgmt Capability in SONiC OS";
+
+    revision 2021-08-09 {
+        description "First Revision";
+    }
+
+    typedef enable-knob {

```

Comment:
@vivekreddynv we already have this type in SONiC:
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-types.yang#L54
```yang
    typedef admin_status {
        type enumeration {
            enum up;
            enum down;
        }
    }
```

Reply:
I feel "enabled|disabled" makes better sense here over "up|down".

Reply:
@vivekreddynv sorry, typo:
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-types.yang#L119
```yang
    typedef admin_mode {
        type enumeration {
            enum enabled;
            enum disabled;
        }
    }
```

Reply:
Oh okay, I'll change this